Libraries/graphics/Sources/Format/PNG.cpp

```diff
@@ -10,19 +10,179 @@
 #include <graphics/Format/PNG.h>
 
 #include <iterator>
+#include <memory>
 #include <ext/optional>
 
 #if _WIN32
+#define CINTERFACE
```

Comment:
I sort of prefer having all of the `#define`s set in code, so the CMake file is less complicated. What's the harm in setting it in the few files that need it?